src/main/scala/thurloe/database/DataAccess.scala

```diff
@@ -20,4 +21,10 @@ case class KeyNotFoundException(userId: String, key: String) extends Exception {
 
 case class InvalidDatabaseStateException(message: String) extends Exception {
   override def getMessage = message
-}
\ No newline at end of file
+}
+
+case class DatabaseConnectionException() extends Exception {
+  override def getMessage = {
+    s"Connection to database was unsuccessful"
+  }
```

Comment:
while I see that other places in the code follow this pattern, I would not override getMessage. Instead pass the message in the constructor to Exception. Can you fix them all?

e.g. KeyNotFoundException should be
case class KeyNotFoundException(userId: String, key: String) extends Exception(s"Key '$key' not found for user '$userId'")